tools/wptserve/wptserve/server.py

```diff
@@ -506,11 +506,13 @@ class H2Headers(dict):
     def __init__(self, headers):
         self.raw_headers = OrderedDict()
         for key, val in headers:
+            key = isomorphic_decode(key)
+            val = isomorphic_decode(val)
             self.raw_headers[key] = val
             dict.__setitem__(self, self._convert_h2_header_to_h1(key), val)
 
     def _convert_h2_header_to_h1(self, header_key):
-        if header_key[1:] in h2_headers and header_key[0] == ':':
+        if header_key[1:] in h2_headers and header_key[:1] == ':':
```

Comment:
I don't think this is necessary if `header_key` is unicode, but I could be wrong. Can you double check?

It was necessary for bytes because `[0]` on a bytes object is a number not a byte-string:

```
>>> x = b'abc'
>>> x[0]
97
```

Reply:
Yes, it's not necessary. Code updated.